debugAdapterProtocol.json

```diff
@@ -2574,6 +2582,10 @@
 							"memoryReference": {
 								"type": "string",
 								"description": "A memory reference to a location appropriate for this result.\nFor pointer type eval results, this is generally a reference to the memory address contained in the pointer.\nThis attribute may be returned by a debug adapter if corresponding capability `supportsMemoryReferences` is true."
+							},
+							"locationReference": {
```

Comment:
>locationReference [](http://example.com/codeflow?start=8&length=17)

suggestion: to make it easier to handle all the 'evaluation results' similarly, might want to call all of these `valueLocationReference`

Reply:
Personally, I prefer the more concise version in the case where the thing it's referring to is unambiguous. It might actually be good that the `Variable` type is different because it does require extra attention to treat the value separately.

Reply:
In that case, maybe use the concise version in all the cases where the concise version is unambiguous (line 2251, 2673) so that the only place it is differentiated is in `Variable`?

Reply:
I would also be in favor of using `valueLocationReference` consistently. In particular, to ensure "future extensibility".

In this particular place in the source code (i.e. for `EvaluateResponse`), it might even make sense to also have a `declarationLocation` (see https://github.com/microsoft/debug-adapter-protocol/pull/494#discussion_r1714204611). I.e., I would retract my [original comment](https://github.com/microsoft/debug-adapter-protocol/issues/343#issuecomment-2274695748)

> Afaict, there is no use case for declarationLocation inside EvaluateResponse, or am I missing something? Similar for OutputEvent

already after only 5 days.

I did not think through the other cases, yet, and I can well imagine that there will be future use cases for a `declarationLocation` also in the other places.

Using `location` instead of `valueLocation` now, would make future evolution a bit awkward, since we might then end up with `location` and `declarationLocation`

Reply:
Thanks for the feedback, I can see it could make sense as a future extensibility point here.

I think in the case of `OutputEvent`, where there is no corresponding data that could be referenced as a "declaration", it would make sense to keep it as `locationReference`, what do you think?

Reply:
I don't think I have ever used a debug-adapter which supports variable references in the `OutputEvent`, so I don't know whether having a `declarationLocation` would be useful there